var/spack/repos/builtin/packages/python/package.py

```diff
@@ -147,12 +160,12 @@ class Python(AutotoolsPackage):
         description='Enable expensive build-time optimizations, if available'
     )
     # See https://legacy.python.org/dev/peps/pep-0394/
-    variant('pythoncmd', default=True,
+    variant('pythoncmd', default=not is_windows,
             description="Symlink 'python3' executable to 'python' "
             "(not PEP 394 compliant)")
 
     # Optional Python modules
-    variant('readline', default=True,  description='Build readline module')
+    variant('readline', default=not is_windows,  description='Build readline module')
```

Comment:
To define the variant on other platforms is there an equivalent to the platform conditional, that allows for negation?

Reply:
Oh good point! We don't have a straightforward means to enable this via the spec (since the spec syntax doesn't have a means to express "everything except x").